### PR TITLE
PROD-709 Network search result count issue fixed

### LINKED ISSUE #3094: Network Search > Post & Page results are not showing properly
**Describe the bug**
If you set "Blog pages show at most" to 5 from Settings > Reading page then in Network Search blog posts result will not show properly. It will show 5 blog post but saying 20 in pagination. Please see this video: https://www.loom.com/share/79f971e81ef24522bad9d6f79dcf80d3

**To Reproduce**
1. "Blog pages show at most" set to 5 from Settings > Reading page.
2. Go to frontend and do a network search with a keyword that has more then 5 blog posts.
3. Click on the Blog Posts from left content category panel.

**Environment**
Demo, Local, Dev, Client - Anywhere

**Support ticket links**
https://secure.helpscout.net/conversation/1659537385/168615/

**Jira issue** : [PROD-709]

[PROD-709]: https://buddyboss.atlassian.net/browse/PROD-709?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by mahedihasannoman**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #3094  .

### How to test the changes in this Pull Request:

1. "Blog pages show at most" set to 5 from Settings > Reading page.
2. Go to frontend and do a network search with a keyword that has more then 5 blog posts.
3. Click on the Blog Posts from left content category panel.
4. See it works fine now.

### Proof Screenshots or Video

https://watch.screencastify.com/v/dRvW1KYPo1qcHgy0slvK

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-search/classes/class-bp-search-posts.php [hunk 1]
```diff
@@ -146,8 +146,10 @@ protected function generate_html( $template_type = '' ) {
 			// lets do a wp_query and generate html for all posts
 			$qry = new WP_Query(
 				array(
-					'post_type' => $this->pt_name,
-					'post__in'  => $post_ids,
+					'post_type' 		=> $this->pt_name,
+					'post__in'  		=> $post_ids,
+					// Override global per page settings and just query with the given IDs
+					'posts_per_page'	=> -1
```
Comment (dipskakadiya) [inline src/bp-search/classes/class-bp-search-posts.php:152]: @mahedihasannoman I don't think this is the correct solution. if will return all post. We can found count on found posts.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

**Comment by dipskakadiya** [comment]
@mahedihasannoman I think without any change you remove the "change requested" tag. Please check my comment.

**Comment by mahedihasannoman** [comment]
> @mahedihasannoman I think without any change you remove the "change requested" tag. Please check my comment.

Hi @dipskakadiya changes added now. Please check. Thanks

**Review by dipskakadiya — DISMISSED** [review]

File: src/bp-search/classes/class-bp-search-posts.php [hunk 2]
```diff
@@ -146,8 +146,10 @@ protected function generate_html( $template_type = '' ) {
 			// lets do a wp_query and generate html for all posts
 			$qry = new WP_Query(
 				array(
-					'post_type' => $this->pt_name,
-					'post__in'  => $post_ids,
+					'post_type' 		=> $this->pt_name,
+					'post__in'  		=> $post_ids,
+					// Override global per page settings and just query with the given IDs
+					'posts_per_page'	=> count( $post_ids )
```
Comment (KartikSuthar) [inline src/bp-search/classes/class-bp-search-posts.php:152]: @mahedihasannoman @dipskakadiya We just need `- 1` here as, already used `post__in`,

```suggestion
					'posts_per_page'	=> - 1
```
```

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

**Review by jitendrabanjara1991 — APPROVED** [review]
LGTM

**Pull request merged**